### DISCUSSION TITLE
Which “Core Instruction” number data is correct?

### DISCUSSION THREAD

**Discussion by hisengberg** [anujpathania/HotSniper#46] [Q&A]
Dear Professor，
    I want to get the total number of instructions spent on each core to complete a program run. 

    Figure 1 is the data obtained through `dumpstats.py` in the python script that comes with HotSniper. Figure 2 is the `sim.out` file automatically generated in each result folder. 

    I found that the final statistics on the number of instructions per core were not the same. The statistics in the two files showed a difference of 100 instructions per core. 

    May I ask why this phenomenon occurs and which data is the real core instruction numbers?

    Hope to get your reply kindly.

Regards,
Walter
![2851701071238_ pic](https://github.com/anujpathania/HotSniper/assets/118739669/ec640cb1-862f-4999-890f-9eae1cf5559b)

![2861701071257_ pic](https://github.com/anujpathania/HotSniper/assets/118739669/b2efe194-d6ff-43c7-ac5c-e902ad5b5fec)

**Comment by anujpathania** [comment]
Hi,

On closer inspection, it seems it is an accounting mismatch inherited from original Sniper.

-- Anuj

**Reply by hisengberg** [reply]
Hi anuj,

      Thank you for your reply. Which data do you think is correct  and which data should I use?

      Hope to get your suggestions.

-- Walter

**Reply by anujpathania** [reply]
I would go with sim.out if you are interested in only accumulated data.

Since difference is quite small, I think the last incomplete scheduling interval is not accounted for in periodic dumpstats.py but is accounted in sim.out. But I am not 100% sure.

-- Anuj

**Reply by hisengberg** [reply]
Thanks a lot.
